src/app/test/matchers/toHaveComputedStyle.ts

```diff
@@ -0,0 +1,76 @@
+import CustomMatcherFactories = jasmine.CustomMatcherFactories;
+import CustomMatcher = jasmine.CustomMatcher;
+import CustomMatcherResult = jasmine.CustomMatcherResult;
+
+/**
+ * Custom matchers use a combination of the following resources:
+ * https://stackoverflow.com/questions/42956195/create-custom-jasmine-matcher-using-typescript
+ * https://jasmine.github.io/tutorials/custom_matcher
+ *
+ * Remember to update matcher-types.d.ts when making changes
+ */
+
+type StyleValue = string | number;
+
+function validateStyles(
+  actual: HTMLElement,
+  expected: Partial<CSSStyleDeclaration>,
+  callback: (
+    key: string,
+    actualStyle: StyleValue,
+    expectedStyle: StyleValue
+  ) => CustomMatcherResult
+): CustomMatcherResult {
+  const computedStyle = getComputedStyle(actual);
+
+  /*
+   * For each expected value, compare to computed value, and return if
+   * miss match detected
+   */
+  Object.entries(expected).forEach(([key, value]) => {
+    const computedValue = computedStyle[key];
+    const result = callback(key, computedValue, value as StyleValue);
+    if (!result.pass) {
+      return result;
```

Comment:
Another question: does this not just return the lambda?

Reply:
Also: 

We either want to return a set of aggregate assertions (more useful IMO)

Or stop running all assertions after the first failure. Currently all assertions are run no matter the fail state which makes the if condition useless.